openssh/Patches/0019-PubKeyPlugin-support.patch

```diff
@@ -119,76 +103,29 @@ index 5d59febc..18a93889 100644
 +		success = (*univ_sym)(pw, type, blob, len);
 +		debug("pubkeyplugin returned: %d", success);
 +		goto out;
++	} else {
++		debug("pubkeyplugin missing %s symbol", UNIV_SYM_NAME);
 +	}
 +
-+	/* Otherwise, continue with the old-style fingerprint symbols. */
-+	fp = sshkey_fingerprint(key, SSH_DIGEST_MD5, SSH_FP_HEX);
-+	if (fp == NULL) {
-+		debug("failed to generate fingerprint");
-+		goto out;
-+	}
-+	if (strncmp(fp, "MD5:", 4) != 0) {
-+		debug("fingerprint not in MD5:hex format");
-+		goto out;
-+	}
-+	/* give the plugin the string without leading MD5: */
-+	argfp = fp + 4;
-+
-+	switch (key->type) {
-+	case KEY_RSA:
-+		rsa_sym = (RSA_SYM)dlsym(handle, RSA_SYM_NAME);
-+		if (rsa_sym == NULL) {
-+			debug("Unable to resolve symbol %s: %s", RSA_SYM_NAME,
-+			dlerror());
-+			goto out;
-+		}
-+		debug2("Invoking %s from %s", RSA_SYM_NAME,
-+		    options.pubkey_plugin);
-+		success = (*rsa_sym)(pw, key->rsa, argfp);
-+		break;
-+	case KEY_ECDSA:
-+		ecdsa_sym = (ECDSA_SYM)dlsym(handle, ECDSA_SYM_NAME);
-+		if (ecdsa_sym == NULL) {
-+			debug("Unable to resolve symbol %s: %s", ECDSA_SYM_NAME,
-+			dlerror());
-+			goto out;
-+		}
-+		debug2("Invoking %s from %s", ECDSA_SYM_NAME,
-+		    options.pubkey_plugin);
-+		success = (*ecdsa_sym)(pw, key->ecdsa, argfp);
-+		break;
-+	default:
-+		debug2("user_key_plugins only support RSA and ECDSA keys");
-+	}
-+
-+	debug("pubkeyplugin returned: %d", success);
-+
 +out:
 +	if (handle != NULL) {
 +		dlclose(handle);
-+		ecdsa_sym = NULL;
-+		rsa_sym = NULL;
 +		univ_sym = NULL;
 +		handle = NULL;
 +	}
 +
 +	if (success) {
-+		verbose("Found matching %s key: %s", sshkey_type(key), fp);
++		verbose("Found matching %s key", sshkey_type(key));
```

Comment:
That had been printing `(null)` since the `sshd_user_key_allowed` was introduced and started being used. The `fp` pointer was no longer being set.

```
Found matching RSA key: (null)
```

I actually went back and captured logs from before the changes to confirm as I first thought it was odd.

Reply:
Gotcha. All good then.